flink-connector-aws/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtilTest.java

```diff
@@ -1021,4 +1021,28 @@ public void testGetV2ConsumerClientProperties() {
                 .containsKey("aws.kinesis.client.user-agent-prefix")
                 .hasSize(2);
     }
+
+    @Test
+    public void testInvalidCustomRecoverableErrorConfiguration() {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(
+                "Provided recoverable exception class could not be found: com.NonExistentExceptionClass");
+
+        Properties testConfig = TestUtils.getStandardProperties();
+        testConfig.setProperty(
+                ConsumerConfigConstants.RECOVERABLE_EXCEPTIONS_PREFIX + "[0].exception",
+                "com.NonExistentExceptionClass");
+
+        KinesisConfigUtil.validateConsumerConfiguration(testConfig);
+    }
+
+    @Test
+    public void testValidCustomRecoverableErrorConfiguration() {
+        Properties testConfig = TestUtils.getStandardProperties();
+        testConfig.setProperty(
+                ConsumerConfigConstants.RECOVERABLE_EXCEPTIONS_PREFIX + "[0].exception",
```

Comment:
should probably add `ConsumerConfigConstants.RECOVERABLE_EXCEPTIONS_EXCEPTION_SUFFIX` but it stutters - open to suggestions

Reply:
Stutters?

Reply:
As in  "ExceptionsExceptionalException..."